cmd/snap-confine/snap-confine.c

```diff
@@ -515,16 +515,47 @@ int main(int argc, char **argv)
 	return 1;
 }
 
-static void enter_classic_execution_environment(void)
+static void enter_classic_execution_environment(sc_invocation * inv)
 {
 	/* 'classic confinement' is designed to run without the sandbox inside the
 	 * shared namespace. Specifically:
-	 * - snap-confine skips using the snap-specific mount namespace
+	 * - snap-confine skips using the snap-specific mount namespace (unless a
+	 parallel installed classic snap)
 	 * - snap-confine skips using device cgroups
 	 * - snapd sets up a lenient AppArmor profile for snap-confine to use
 	 * - snapd sets up a lenient seccomp profile for snap-confine to use
 	 */
+
 	debug("skipping sandbox setup, classic confinement in use");
+
+	/* Parallel installed classic snap get special handling */
+	if (!sc_streq(inv->snap_instance, inv->snap_name)) {
+		/* Construct a mount namespace where the snap instance directories are
+		 * visible under the regular snap name. In order to do that we will:
+		 *
+		 * - convert SNAP_MOUNT_DIR into a mount point
+		 * - convert /var/snap into a mount point
+		 * - set slave propagation for both
```

Comment:
if this list is meant to be in-order (which I think it should be), then this item should be below the `- create a new mount namespace` item below I think